nextjs-13/src/app/page.tsx

```diff
@@ -1,20 +1,36 @@
 'use client';
-import { useState } from 'react';
+import { Suspense, useEffect, useState } from 'react';
+import { ClipLoader } from 'react-spinners';
 
 import Head from 'next/head';
 import { redirect } from 'next/navigation';
 import { useSession } from 'next-auth/react';
 
 import CreateNewsletterModal from '@components/CreateNewsletterModal';
+import ListNewsletter from '@components/ListNewsletter';
+import requestManager from 'lib/request/manager';
+import promiseWrapper from 'lib/request/promiseWrapper';
 
 const Home = () => {
   const { status } = useSession();
   const [modalIsOpen, setIsOpen] = useState(false);
+  const [promise, setPromise] = useState();
 
-  if (status === 'loading') {
-    return <p>Loading...</p>;
+  async function fetchRecords() {
+    const res = await requestManager('GET', 'v1/newsletter');
```

Comment:
Ditto from below. Avoid abbreviations: https://nimblehq.co/compass/development/code-conventions/#convention-12

```suggestion
    const newsletterList = await requestManager('GET', 'v1/newsletter');
```

Reply:
Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/110/commits/d30f9f526aeaccaa44ac7bfcc866291c1952f753#diff-223e9b714cf2c9b5f163079f0bd63d3a6eac9c46915fc3afde20a2742897d706R20)